core/src/accounts_hash_verifier.rs

```diff
@@ -155,17 +179,53 @@ impl AccountsHashVerifier {
             }
         }
 
+        cluster_info.push_accounts_hashes(hashes.clone());
+    }
+
+    fn handle_snapshot(
+        accounts_package: AccountsPackage,
+        pending_snapshot_package: Option<&PendingSnapshotPackage>,
+        snapshot_config: Option<&SnapshotConfig>,
+    ) {
         if let Some(snapshot_config) = snapshot_config {
-            if snapshot_package.block_height % snapshot_config.full_snapshot_archive_interval_slots
-                == 0
-            {
-                if let Some(pending_snapshot_package) = pending_snapshot_package {
+            if let Some(pending_snapshot_package) = pending_snapshot_package {
+                if accounts_package.block_height
+                    % snapshot_config.full_snapshot_archive_interval_slots
+                    == 0
+                {
+                    let snapshot_package = SnapshotPackage::new(
+                        accounts_package,
+                        snapshot_config.snapshot_package_output_path.clone(),
+                        snapshot_config.archive_format,
+                        None,
+                    );
+
                     *pending_snapshot_package.lock().unwrap() = Some(snapshot_package);
-                }
-            }
-        }
+                } else if accounts_package.block_height
+                    % snapshot_config.incremental_snapshot_archive_interval_slots
+                    == 0
+                {
+                    let last_full_snapshot_slot =
+                        snapshot_utils::get_highest_full_snapshot_archive_slot(
+                            &snapshot_config.snapshot_package_output_path,
+                        );
 
-        cluster_info.push_accounts_hashes(hashes.clone());
+                    match last_full_snapshot_slot {
+                        None => warn!("No full snapshot archive found; will not process incremental snapshot package!"),
+                        Some(incremental_snapshot_base_slot) => {
+                            let snapshot_package = SnapshotPackage::new(
+                                accounts_package,
+                                snapshot_config.snapshot_package_output_path.clone(),
+                                snapshot_config.archive_format,
+                                Some(incremental_snapshot_base_slot),
+                            );
+
+                            *pending_snapshot_package.lock().unwrap() = Some(snapshot_package);
```

Comment:
If the last full snapshot hasn't been ingested yet, maybe we don't want to overwrite it with an incremental snapshot?

Reply:
Eek! Yeah I agree. Will fix!

Reply:
Done!

Reply:
Maybe a. test for `submit_for_packaging()`?